streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -110,6 +127,7 @@ public void flush() {
     @Override
     public void close() {
         producer.close();
+        checkForException();
```

Comment:
Hmm, yes it is only called from the test i wrote. The producer is closed during `shutdown` in `StreamThread`. Which i don't think would matter on its own, but it is semi-related to your comment above. During `shutdown` and `suspend` we should only commit offsets after we've flushed and/or closed the producer - otherwise we run the risk of violating at-least-once.